### PR TITLE
Feature/dorsh/dropdown

### PR DESCRIPTION
**Pull request by DorShakedMonday**
Description: **New component - Dropdown**

- Added Dropdown component based on react-select, react-virtualized and react-select-virtualized.
- Fixed Icon props naming
- Fixed FieldLabel usage of Icon component

#### Basic
- [x] Used plop (npm run plop) to create a new component. This command will create a folder for your component and will update index.js file
- [x] PR has description
- [x] New component is functional and using React Hooks. 
- [x] Component defines PropTypes
#### Style
- [x] Styles are added to NewComponent.scss file inside of the NewComponent folder
- [x] CSS selectors are named using [BEM convention](http://getbem.com/naming/) 
- [x] Design is compatible with [Monday Design System](https://design.monday.com/)
#### Storybook
- [x] Stories were added to /src/NewComponent/__stories__/NewComponent.stories.js file
- [x] Stories include sandbox story with knobs for each property
- [x] Stories include all flows of using the component
- [x] Component passed Accessibility Plugin checks
#### Tests
- [ ] Tests are compliant with TESTING_README.md instructions

### CODE REVIEW

**Review by orrgottlieb — CHANGES_REQUESTED** [review]
Awesome awesome awesome PR.
I have some requests but overall amazing PR
Thank you

File: src/components/Dropdown/Dropdown.jsx [hunk 1]
```diff
@@ -0,0 +1,300 @@
...
+          "dropdown-wrapper__menu--close": !isOpen
+        })}
+      >
+        {props.children}
+      </components.Menu>
+    ),
+    [isOpen]
+  );
+
+  const DropdownIndicator = useCallback(
```
[the remark this answers is not in the document]
Comment (DorShakedMonday) [reply src/components/Dropdown/Dropdown.jsx:68]: Done

File: src/components/Dropdown/Dropdown.jsx [hunk 2]
```diff
@@ -0,0 +1,300 @@
...
+          iconSize={getIndicatorSize(size)}
+          tabindex="-1"
+        />
+      </components.ClearIndicator>
+    ),
+    [size]
+  );
+
+  let DropDownComponent;
+  if (isVirtualized) {
```
[the remark this answers is not in the document]
Comment (DorShakedMonday) [reply src/components/Dropdown/Dropdown.jsx:132]: We discussed this and decided to leave it this way and maybe separate later.

File: src/components/Dropdown/Dropdown.jsx [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+  onKeyDown: NOOP,
+  onFocus: NOOP,
+  searchable: true,
+  options: [],
+  noOptionsMessage: NOOP,
+  clearable: true,
+  size: SIZE.MEDIUM
+};
+
+Dropdown.propTypes = {
```
[the remark this answers is not in the document]
Comment (DorShakedMonday) [reply src/components/Dropdown/Dropdown.jsx:200]: XD

File: src/components/Dropdown/Dropdown.styles.js [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+};
+
+const getColor = () => {
+  const color = getCSSVar("primary-text-color");
+  const backgroundColor = getCSSVar("primary-background-color");
+  return { color, backgroundColor };
+};
+
+const getFont = size => {
+  switch (size) {
```
[the remark this answers is not in the document]
Comment (DorShakedMonday) [reply src/components/Dropdown/Dropdown.styles.js:41]: Mistake, the code was different at the start hhh

File: src/components/Dropdown/Dropdown.scss [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+      @include theme-prop(background-color, primary-selected-color);
+      cursor: pointer;
+    }
+  }
+}
+
+@keyframes fadeOut {
+  0% {
+    opacity: 1;
+    max-height: 300px;
```
[the remark this answers is not in the document]
Comment (DorShakedMonday) [reply src/components/Dropdown/Dropdown.scss:66]: Done

File: src/components/Dropdown/Dropdown.styles.js (hunk 4 above)
[the remark this answers is not in the document]
Comment (DorShakedMonday) [reply src/components/Dropdown/Dropdown.styles.js:41]: Done

File: src/components/FieldLabel/FieldLabel.jsx [hunk 6]
```diff
@@ -22,7 +22,7 @@ const FieldLabel = forwardRef(
     return (
       <section className="label-component--wrapper">
         <Icon
-          iconName={iconName}
+          icon
```
Comment (orrgottlieb) [inline src/components/FieldLabel/FieldLabel.jsx:25]: should have the icon passed to the icon component

File: src/components/Icon/Icon.jsx [hunk 7]
```diff
@@ -47,7 +48,7 @@ const Icon = forwardRef(
         <IconComponent
           size={iconSize.toString()}
           onClick={onClick}
-          tabIndex={tabindex}
+          tabIndex={externalTabIndex || tabindex}
```
Comment (orrgottlieb) [inline src/components/Icon/Icon.jsx:51]: this could be an issue if the external is 0 and the tabIndex might be for example -1 ?

**Review by orrgottlieb — APPROVED** [review]

**Pull request merged**